operator/rust/crates/operator/src/start_operator.rs

```diff
@@ -140,118 +132,115 @@ async fn monitor_new_tasks() -> Result<()> {
+async fn register_operator() -> Result<()> {
+    let signer = PrivateKeySigner::from_str(&KEY.clone())?;
+    let wallet = EthereumWallet::from(signer.clone());
+    let pr = ProviderBuilder::new()
+        .with_recommended_fillers()
+        .wallet(wallet)
+        .on_http(Url::from_str(&ANVIL_RPC_URL)?);
 
     let default_slasher = Address::ZERO; // We don't need slasher for our example.
     let default_strategy = Address::ZERO; // We don't need strategy for our example.
 
+    let data = std::fs::read_to_string("contracts/deployments/core/31337.json")?;
+    let el_parsed: EigenLayerData = serde_json::from_str(&data)?;
+    let delegation_manager_address: Address = el_parsed.addresses.delegation.parse()?;
+    let avs_directory_address: Address = el_parsed.addresses.avs_directory.parse()?;
+
     let elcontracts_reader_instance = ELChainReader::new(
-        logger.clone(),
+        get_logger().clone(),
         default_slasher,
-        delegation_manager_contract_address,
-        avs_directory_contract_address,
-        RPC_URL.clone(),
+        delegation_manager_address,
+        avs_directory_address,
+        ANVIL_RPC_URL.to_string(),
     );
     let elcontracts_writer_instance = ELChainWriter::new(
-        delegation_manager_contract_address,
+        delegation_manager_address,
         default_strategy,
+        Address::ZERO,
         elcontracts_reader_instance.clone(),
-        RPC_URL.clone(),
+        ANVIL_RPC_URL.to_string(),
         KEY.clone(),
     );
 
-    let operator = Operator::new(
-        wallet.address(),
-        wallet.address(),
-        Address::ZERO,
-        0u32,
-        None,
-    );
+    let operator = Operator {
+        address: signer.address(),
+        earnings_receiver_address: signer.address(),
+        delegation_approver_address: Address::ZERO,
+        staker_opt_out_window_blocks: 0u32,
+        metadata_url: None,
+    };
 
-    // let is_registered = elcontracts_reader_instance
-    //     .is_operator_registered(wallet.address())
-    //     .await
-    //     .unwrap();
-    // logger
-    //     .tracing_logger
-    //     .as_ref()
-    //     .unwrap()
-    //     .info(&format!("is registered {}", is_registered), &[""]);
-    #[allow(unused_doc_comments)]
-    ///In case you are running holesky. Comment the below register_as_operator call after the first
-    /// call . Since we can register only once per operator.
-    let _tx_hash = elcontracts_writer_instance
+    let is_registered = elcontracts_reader_instance
+        .is_operator_registered(signer.address())
+        .await
+        .unwrap();
+    get_logger().info(&format!("is registered {}", is_registered), &"");
+    #[allow(unused)]
+    let tx_hash = elcontracts_writer_instance
         .register_as_operator(operator)
-        .await;
-    logger
-        .tracing_logger
-        .as_ref()
-        .unwrap()
-        .info("Operator registered on EL successfully", &[""]);
+        .await?;
+    get_logger().info(
```

Comment:
I think there might be something wrong here, at least when I run it under my system. This log on my system prints:
```
2024-10-31T14:31:23.926333Z  INFO eigen_logging::tracing_logger: Operator registered on EL successfully tx_hash {tx_hash:?} [""]
```

When I start the operator, the operator immediately ends, and perhaps is related to this.   Here is my complete output:
```cargo run --bin start_operator
    Finished `dev` profile [unoptimized + debuginfo] target(s) in 0.22s
     Running `target/debug/start_operator`
2024-10-31T14:31:23.910989Z  INFO eigen_logging::tracing_logger: is registered true [""]
2024-10-31T14:31:23.911014Z  INFO eigen_client_elcontracts::writer: registering operator 0xf39fd6e51aad88f6f4ce6ab8827279cfffb92266 to EigenLayer
2024-10-31T14:31:23.926303Z  INFO eigen_client_elcontracts::writer: failed to register operator tx_hash=0x7bf118c2d9105c6ce7b736127bba8392b7d79f92fba25d3e8c3173325f902a68
2024-10-31T14:31:23.926333Z  INFO eigen_logging::tracing_logger: Operator registered on EL successfully tx_hash {tx_hash:?} [""]
2024-10-31T14:31:23.943066Z  INFO eigen_logging::tracing_logger: Operator registered on AVS successfully :0xf39Fd6e51aad88F6F4ce6aB8827279cffFb92266 , tx_hash :0xe054cc9ddaa9f15f01835b325dfa182a4a84f827bebbbc68cead46e1ec44e180 [""]
```

Reply:
it should work now